core/src/main/scala/sttp/tapir/Tapir.scala

```diff
@@ -398,7 +398,14 @@ trait Tapir extends TapirExtensions with TapirComputedInputs with TapirStaticCon
     */
   def oneOfBody[T](first: EndpointIO.Body[_, T], others: EndpointIO.Body[_, T]*): EndpointIO.OneOfBody[T, T] =
     EndpointIO.OneOfBody[T, T](
-      (first +: others.toList).map(b => EndpointIO.OneOfBodyVariant(ContentTypeRange.exactNoCharset(b.codec.format.mediaType), b)),
+      (first +: others.toList).map(b => EndpointIO.OneOfBodyVariant(ContentTypeRange.exactNoCharset(b.codec.format.mediaType), Left(b))),
+      Mapping.id
+    )
+
+  /** Streaming variant of [[oneOfBody]]. */
+  def oneOfBody[T](first: EndpointIO.StreamBodyWrapper[_, T], others: EndpointIO.StreamBodyWrapper[_, T]*): EndpointIO.OneOfBody[T, T] =
```

Comment:
You probably already explained it to me in the original issue but why not accept `StreamBodyIO` as input and do the `.toEndpointIO` inside this method?

Though it might be safer this way as users are given a hint that if they have to do `.toEndpointIO` it's probably because there's something special about it.

Reply:
exactly for this reason :) as we are bypassing some of the typesafety mechanisms, I wanted to make it explicit. Maybe people will even read the scaladoc of `.toEndpointIO` :)